reasoner/controller/ConcludableController.java

```diff
@@ -241,22 +234,44 @@ private void mayRequestConnection(REQ conclusionRequest) {
 
         public static class Match extends Processor<Map<Variable, Concept>, ConceptMap, Match.Request, Match> {
 
-            private final Concludable concludable;
-
             Match(
                     Driver<Match> driver, Driver<ConcludableController.Match> controller, Concludable concludable,
                     Context context, ConceptMap bounds, Set<Variable.Retrievable> unboundVars,
                     Map<Conclusion, Set<Unifier>> conclusionUnifiers,
                     Supplier<FunctionalIterator<ConceptMap>> traversalSuppplier, Supplier<String> debugName
             ) {
-                super(driver, controller, context, bounds, unboundVars, conclusionUnifiers, traversalSuppplier,
-                      debugName);
-                this.concludable = concludable;
+                super(driver, controller, concludable, context, bounds, conclusionUnifiers, traversalSuppplier, debugName);
             }
 
+
             @Override
-            protected void mayAddTraversal() {
-                new Source<>(this, traversalSuppplier).flatMap(this::filterInferred).registerSubscriber(hubReactive());
+            public void setUp() {
+                setHubReactive(fanInFanOut(this));
+                if (canBypassReasoning(bounds)) {
+                    addTraversal(true);
+                } else {
+                    addTraversal(false);
+                    addRules();
+                }
+            }
+
+            private boolean canBypassReasoning(ConceptMap bounds) {
+                if (concludable.isHas()) {
+                    Concept owner; Concept attribute;
+                    return  (owner = bounds.get(concludable.asHas().owner().id())) != null &&
+                            (attribute = bounds.get(concludable.asHas().attribute().id())) != null &&
+                            (owner.asThing().hasInferred(attribute.asAttribute()) || owner.asThing().hasNonInferred(attribute.asAttribute()));
```

Comment:
We check the existence of the edge here, to determine whether we can bypass traversal.
Will see how hard it is to bypass the full traversal for non-relation cases.